app/grants/models.py

```diff
@@ -272,6 +272,50 @@ def org_profile(self):
             return profiles.first()
         return None
 
+    @property
+    def history_by_month(self):
+        # gets the history of contributions to this grant month over month so they can be shown o grant details
+        # returns [["", "Subscription Billing",  "New Subscriptions", "One-Time Contributions", "CLR Matching Funds"], ["December 2017", 5534, 2011, 0, 0], ["January 2018", 10396, 0 , 0, 0 ], ... for each monnth in which this grant has contribution history];
+        CLR_PAYOUT_HANDLES = ['vs77bb', 'gitcoinbot', 'notscottmoore', 'owocki']
+        month_to_contribution_numbers = {}
+        for sub in self.subscriptions.all():
```

Comment:
`prefetch_related('subscription_contribution')` will save us a bunch of DB lookups I think. Then filter `success=True` in python like `for contrib in [sc for sc in sub.subscription_contribution if sc.success=True]`. What do you think? Cost some CPU but greatly reduces DB lookups so should be faster to execute.

Reply:
hows this? i tested locally 3f49ec63a11fabc1f359aea8e4123b83900a6fc6